cypress/support/MonitorDocumentChanges.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.  
+ * Licensed under the MIT License.
+*/
+
+const helpers = require('../support/helpers.js')
+
+var MonitorDocumentChanges = (function()
+{
+    var lastMonitorTime = 0
+    var lastChangeTime = 0
+    var canRefreshTrainingStatusTime = 0
+    var lastHtml
+    var StopLookingForChanges = false
+    var dumpHtml = false;
+    
+    var expectingSpinner
+    var currentSpinnerText = ''
+
+    // Initialize this only once.
+    //   It gets called the first time it is imported, which should be at or near the top of the index.js file
+    //   Then it potentially could get called additional times when the require statement is used.
+    //   BUGBUG: this initialize once logic does not work because lastChangeTime gets zeroed everytime this is imported or required.
+    if (lastChangeTime == 0) initialize();
```

Comment:
So initialize is getting called every time?   Does BUGBUG need fixing or does it not matter given the way the test work?

Reply:
It used to matter, but now in this case it does not matter. I left the comment in to remind me to deal with this issue because I do need to eventually figure this out. But now I clearly remember this problem and will face it in the future when I have to in another piece of code.

Reply:
I fixed the code and removed the comment.